File: Java/src/No1078.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

import java.util.ArrayList;
import java.util.List;

public class No1078 {
    public String[] findOcurrences(String text, String first, String second) {
        // 首先按照 " " 空格去切分字符串
        String[] words = text.split(" ");
        List<String> list = new ArrayList<>();
        // 然后按照规则直接检查就可以了
        for (int i = 0; i < words.length - 2; ++i) {
            if (words[i].equals(first) && words[i + 1].equals(second))
                // 遇到符合要求的，直接添加进返回值
                list.add(words[i + 2]);
        }
        String[] ans = new String[list.size()];
        list.toArray(ans);
        return ans;
    }
}

File: CPP/No1078.cc
/**
 * Created by Xiaozhong on 2021/12/26.
 * Copyright (c) 2021/12/26 Xiaozhong. All rights reserved.
 */
#include "iostream"
#include "vector"
#include "string"

using namespace std;

class Solution {
private:
    // 按照指定字符切分字符串
    vector<string> split(const string &s, const char delimiter) {
        vector<string> ret;
        string buf;
        for (const char &c: s) {
            if (c != delimiter) buf.push_back(c);
            else {
                ret.emplace_back(buf);
                buf = "";
            }
        }
        if (!buf.empty()) ret.emplace_back(buf);
        return ret;
    }

public:
    vector<string> findOcurrences(string text, string first, string second) {
        vector<string> words = split(text, ' ');
        vector<string> ans;
        for (int i = 2; i < words.size(); ++i) {
            if (words[i - 2] == first && words[i - 1] == second) ans.emplace_back(words[i]);
        }
        return ans;
    }
};